src/BugsnagUnity.mm

```diff
@@ -195,19 +217,16 @@ void bugsnag_clearEventMetadataSection(const void *event, const char *section){
    [((__bridge BugsnagEvent *)event) clearMetadataFromSection:@(section)];
 }
 
-void bugsnag_setEventMetadata(const void *event, const char *tab, const char *metadata[], int metadataCount) {
-  if (tab == NULL)
-    return;
+void bugsnag_setEventMetadata(const void *event, const char *tab, const char *metadataJson) {
+  
+    if (tab == NULL || metadataJson == NULL)
+    {
+        return;
+    }
 
-  NSString *tabName = [NSString stringWithUTF8String: tab];
-  NSMutableDictionary *ns_metadata = [NSMutableDictionary new];
+    NSString *tabName = [NSString stringWithUTF8String: tab];
 
-  for (int i = 0; i < metadataCount; i += 2) {
-    if (metadata[i] && metadata[i+1]) {
-        ns_metadata[@(metadata[i])] = @(metadata[i+1]);
-    }
-  }
-  [((__bridge BugsnagEvent *)event) addMetadata:ns_metadata toSection:tabName];
+    [((__bridge BugsnagEvent *)event) addMetadata:getDictionaryFromMetadataJason(metadataJson) toSection:tabName];
```

Comment:
Pedantic: prefer `@()` to `[NSString stringWithUTF8String:]`

```suggestion
    [((__bridge BugsnagEvent *)event) addMetadata:getDictionaryFromMetadataJason(metadataJson) toSection:@(tab)];
```

Reply:
Done 👍